W-14955232-MMPmuleVersion-dm (#2623)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/_partials/mmp-deploy-to-rtf.adoc

If not set, defaults to +https://anypoint.mulesoft.com+. | No
| `muleVersion` | The Mule runtime engine version to run in your Runtime Fabric instance.  +
Ensure that this value is equal to or higher than the earliest required Mule version of your application. +
Example values: `4.6.0` selects the last build where it defaults to Edge channel and Java 8. If you want to select a different channel or Java version, specify the full Mule version name as in `4.6.0:1e-java17`. +

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6.0</muleVersion>`. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes

Author: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, the `muleVersion` property doesn't allow you to specify the release channel and Java version which may cause unexpected behavior. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes
```

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes

Reviewer: I think this example is only valid for cloudhub 1 and it may be miss leading for RTF and CH2. AFAIK with older versions of MMP only the base version is accepted (i.e. 4.6.0). And, as MMP selects the latest tag for that base version the end result may be unexpected too as it may select a tag from any release channel and the tag may be java8 or java17, depending on which one was selected.

In summary, I would replace `4.6-lts-java17` with `4.6.0` and I would not provide any guideline of what would be the end result as it may vary each time depending on what was the latest published image.

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes

Author: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6.0</muleVersion>`. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes
```

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`lts` or `e`) and `Java` (`java8` or `java17`) properties to define deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes

Reviewer 2: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`lts` or `e`) and `Java` (`java8` or `java17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes]. | Yes
```

Reviewer 2: or "to select the deployment"?

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`lts` or `e`) and `Java` (`java8` or `java17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes

Author: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`NONE`, `EDGE`, or `LTS`) and `javaVersion` (`8` and `17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes
```

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`NONE`, `EDGE`, or `LTS`) and `javaVersion` (`8` or `17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes

Author: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection may cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0. See xref:release_notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven Plugin 4.1.0 Release Notes].  | Yes
```

> When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`NONE`, `EDGE`, or `LTS`) and `javaVersion` (`8` and `17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes

Author: ```suggestion
When using Mule Maven plugin versions 3.8.0 and 4.0.0 for deployments, using the `muleVersion` property to specify the release channel and Java version selection might cause unexpected deployments. For example, specifying `<muleVersion>4.6-lts-java17</muleVersion>` in the plugin 3.8.0 or 4.0.0 deploys the application with an Edge release channel and a Java 8 selection. To specify the release channel and Java version, upgrade the plugin to version 4.1.0 and use the `releaseChannel` (`NONE`, `EDGE`, or `LTS`) and `javaVersion` (`8` or `17`) properties to define the deployment selection. See xref:release-notes::mule-maven-plugin/mule-maven-plugin-4.1.0-release-notes.adoc[Mule Maven plugin 4.1.0 Release Notes]. | Yes
```
